workspace_tools/toolchains/gcc.py

```diff
@@ -183,7 +183,8 @@ def __init__(self, target, options=None, notify=None, macros=None, silent=False,
         GCC.__init__(self, target, options, notify, macros, silent, GCC_ARM_PATH, extra_verbose=extra_verbose)
 
         # Use latest gcc nanolib
-        self.ld.append("--specs=nano.specs")
+        if "small" in self.options:
```

Comment:
I had this discussion with Russ and hopefully it's not too out of place here. I believe anything that is a required option (like target and toolchain) should not cause behavior changes in your code. Meaning, if you switch to the proposed nGCC_ARM toolchain, if you were using multiple threads this would directly impact your  code's stability.

I think adding the option to enable the "small" lib is the way to go. That way if I know about it and understand its implications I will turn it on. Otherwise I'll assume the defaults for a toolchain are safe to use in all cases.